src/content/guides/farmlands-pos-integration.mdoc

```diff
@@ -382,41 +386,48 @@ autonumber
 
 participant Cardholder
 participant POS
+participant POS Server
 participant Centrapay
 
 Note over Cardholder: Present Barcode
 Cardholder->>POS: Scan Barcode
 
-POS->>Centrapay: Create Payment Request with Barcode and the preAuth flag
+POS->>POS Server: Request Payment
+
+POS Server->>Centrapay: Create Payment Request with Barcode and the preAuth flag
 
 loop Poll for Payment Confirmation
-	POS->>Centrapay: Get Payment Request
+	POS Server->>Centrapay: Get Payment Request
 
-	alt Payment Condition Pending
-			note over POS: ❓ Prompt Cashier to Accept or Decline Condition
-			POS->>Centrapay: POS Accepts Condition
-		else Payment Request Status is paid
-			Note over POS: Stop Polling for Confirmation
-		end
+  opt Payment Condition Pending
+    note over POS: ❓ Prompt Cashier to Accept or Decline Condition
+    POS->>Centrapay: POS Accepts Condition
+  end
 end
 
+POS Server->>POS: Send Payment Status
+
 Note over POS: ✅ Display Authorization Confirmation
 
-Note over POS, Centrapay: ⏱ Some time later..
+Note over Cardholder, Centrapay: ⏱ Some time later..
 
 loop 0..Many (Up to Authorised Amount)
-	POS->>Centrapay: Make Confirmations Against Authorised Funds
-
+  POS->>POS Server: Request Confirmation
+	POS Server->>Centrapay: Make Confirmations Against Authorised Funds
+  POS Server->>POS: Send Confirmation Status
```

Comment:
Question: Is the POS server pushing a notification here or is it the response to line 415? If it is the response, I think this would be better like:

```
POS->>POS Server: Make Confirmations Against Authorised Funds
POS Server->>Centrapay: Make Confirmations Against Authorised Funds
Centrapay-->>POS Server: Response
POS Server-->>POS: Response
```